src/compile.rs

```diff
@@ -470,11 +472,15 @@ impl Compiler {
                         }
                     };
 
-                    let cache = core::cell::RefCell::new(dfa.create_cache());
+                    let create: CachePoolFn = alloc::boxed::Box::new({
+                        let dfa = dfa.clone();
+                        move || dfa.create_cache()
+                    });
+                    let cache_pool = Pool::new(create);
```

Comment:
I don't like that it clones the DFA just to create a cache for it. Do you think it would make sense to deserialize it or is there a better way? https://docs.rs/regex-automata/0.4.13/regex_automata/dfa/index.html#example-deserialize-a-dfa